builds/e2e/e2e.yaml

```diff
@@ -48,19 +56,86 @@ jobs:
     - template: templates/e2e-run.yaml
 
 ################################################################################
-  - job: linux_amd64
+  - job: ubuntu_1804_msmoby
+################################################################################
+    displayName: Ubuntu 18.04 with iotedge-moby
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
+
+    variables:
+      os: linux
+      arch: amd64
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
```

Comment:
Is there a way in DevOps to declare these hardcoded values without exposing them as a pipeline variable? I have been burned before where I have declared a hardcoded value that used to be configurable in an old branch, and thus the pipeline still had a configurable variable that would overwrite this hardcoded value.

Reply:
Admittedly this turned into a bit more than just a cherry-pick, but I agree this should probably be a separate effort. And these are variables not parameters...does that help?

Reply:
My understanding is this:
1. Pipeline variables can be set from the pipeline, but also declared like this as hardcoded strings. This leads to the problem I was mentioning above.
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-variables-in-pipeline
2. The yaml can also have parameters so that the caller-yaml can pass values to the callee-yaml. These are accessed with different syntax than the pipeline variables.

Reply:
Sorry I got the wrong ideas in my head about parameters vs variables. Back on track now. I don't know off-hand how to solve the problem you're describing, but we can keep thinking about it...

Reply:
As an alternative, we could declare these hardcoded values in `parameters:` header at the top of file. But seems a bit weird because nothing calls this yaml. Not a perfect solution but avoids the problem.